connection_go18.go

```diff
@@ -0,0 +1,234 @@
+// Go MySQL Driver - A MySQL-Driver for Go's database/sql package
+//
+// Copyright 2012 The Go-MySQL-Driver Authors. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+)
+
+type setfinish interface {
```

Comment:
-1
because `setFinish` doesn't finalize,  but the function `f` does.
Another suggestion.

``` go
type setterFinalizer interface {
    setFinalizer(finalizer func())
}
```

Are there other better names?